tests/pgautofailover_utils.py

```diff
@@ -484,13 +492,36 @@ def show_uri(self, json=False):
         """
         Runs pg_autoctl show uri
         """
-        command = PGAutoCtl(self.vnode, self.datadir)
+        command = PGAutoCtl(self)
         if json:
             out, err = command.execute("show uri", 'show', 'uri', '--json')
         else:
             out, err = command.execute("show uri", 'show', 'uri')
         return out
 
+    def detailed_exception(self, error_msg, other_node=None):
```

Comment:
I think this could be better named `get_detailed_exception` so that we don't have to second-guess the choice of returning the object compared to raising the exception. Or maybe `build_detailed_exception` or even `make_detailed_exception`?

Reply:
Reading it another time, it should return only the exception error message. So what about `build_detailed_error_message` or `make_error_report` or something like that?